src/gql/server/Buttons.tsx

```diff
@@ -1,72 +1,76 @@
-import { DocumentNode, gql, useMutation } from "@apollo/client"
-import { useContext, useEffect, useRef } from "react"
-import { Button } from "primereact/button"
-import { ButtonStateType, TargetType } from "@/types"
-import { VariablesContext } from "@Contexts/Variables/VariablesProvider"
-import { Toast } from "primereact/toast"
-import { BTN_CLASSES } from "@/Helpers/constants"
+import { useMutation, OperationVariables } from '@apollo/client';
+import { VariablesOf, TypedDocumentNode } from '@graphql-typed-document-node/core';
+
+import { useContext, useEffect, useRef } from 'react';
+import { Button } from 'primereact/button';
+import { ButtonStateType } from '@/types';
+import { VariablesContext } from '@Contexts/Variables/VariablesProvider';
+import { Toast } from 'primereact/toast';
+import { BTN_CLASSES } from '@/Helpers/constants';
+import { graphql } from './gen';
+import { Instrument, SlewOptionsInput } from './gen/graphql';
 
 // Generic mutation button
-function MutationButton({
+function MutationButton<A, B, T extends TypedDocumentNode<A, B>>({
```

Comment:
This is an example of a bit more than just adding types. This component now accepts typed documents, so it typechecks the variables for the mutations. E.g. one of the uses of this component was using a string instead of number for a variable.

Reply:
I understand the idea, but I am not familiarized with the syntax.
In this case you are extending the TypedDocumentNode to receive a generic B type, but what about that A generic type, where is that coming from?
Actually I will go study a bit more about generics in typescript.

Reply:
This function has 3 generic type arguments: `A`, `B` and `T`. 

`T` has a constraint that whatever `T` is, it _has_ to extend `TypedDocumentNode`, which has 2 generic arguments (for the Variables, and the Result). We fill those with `A` and `B`. 

In this case, the `A` and `B` are not really that important, but we have to ask for them first to put them in `TypedDocumentNode`. But since we now have an `A` (`Variables`) we use it as the type for the variables passed to the function. 

Ideally it would be something like `T extends TypedDocumentNode<infer A, infer B>`, but that does not exist in typescript